tests/test_core.py

```diff
@@ -426,8 +426,42 @@ def test_as_graph_duplicate_args(self):
 
 
 class TestSanityCheck(unittest.TestCase):
-    # TODO: We need to test all of sanity check
-    pass
+    def setUp(self):
+        pyrtl.reset_working_block()
+
+    def sanity_error(self, msg):
+        with self.assertRaisesRegexp(pyrtl.PyrtlError, msg):
+            pyrtl.working_block().sanity_check()
+
+    def test_missing_bitwidth(self):
+        inp = pyrtl.Input()
+        out = pyrtl.Output(8)
+        self.sanity_error("missing bitwidth")
+
+    def test_duplicate_names(self):
+        inp = pyrtl.Input(8, 'hi')
+        out = pyrtl.Output(8, 'hi')
+        out <<= inp
+        self.sanity_error("Duplicate wire names")
+
+    def test_unknown_wires(self):
+        inp = pyrtl.Input(8, 'inp')
+        out = pyrtl.Output(8, 'out')
+        out <<= inp
+        pyrtl.working_block().wirevector_set.discard(inp)
+        with self.assertRaises(pyrtl.PyrtlInternalError):  # sanity_check_net()
+            self.sanity_error("Unknown wires")
```

Comment:
Um yeah `sanity_check_net()` throws one and then later on in `sanity_check()` another one is thrown.